### PR TITLE
EPMEDU-597: Feeding booking

### PR DESCRIPTION
**Pull request by igorTimofiejczyk**
Added background support for feeding flow
PR closes: 
EPMEDU-603
EPMEDU-597
EPMEDU-796

### CODE REVIEW

File: animeal/src/Flows/Main/Modules/Home/Main/HomeViewModel.swift [hunk 1]
```diff
@@ -53,23 +57,14 @@ final class HomeViewModel: HomeViewModelLifeCycle, HomeViewInteraction, HomeView
...
-                                self.onRouteRequestHaveBeenPrepared?(pointItemView.coordinates)
-                            }
-                        } catch {
-                            DispatchQueue.main.async {
-                                self.onErrorHaveBeenPrepared?(error.localizedDescription)
-                                self.fetchFeedingPoints()
-                            }
-                        }
-                    }
+                    self.onRouteRequestHaveBeenPrepared?(
```
Comment (diantyn) [inline animeal/src/Flows/Main/Modules/Home/Main/HomeViewModel.swift]: weak self

File: animeal/src/Flows/Main/Modules/Home/Main/HomeViewModel.swift [hunk 2]
```diff
@@ -86,8 +81,54 @@ final class HomeViewModel: HomeViewModelLifeCycle, HomeViewInteraction, HomeView
             let action = model.fetchFeedingAction(request: .autoCancelFeeding)
             onFeedingActionHaveBeenPrepared?(feedingActionMapper.mapFeedingAction(action))
         case .confirmCancelFeeding:
-            load()
-            model.processCancelFeeding()
+            Task {
+                do {
```
Comment (diantyn) [inline animeal/src/Flows/Main/Modules/Home/Main/HomeViewModel.swift:86]: weak self

File: animeal/src/Flows/Main/Modules/Home/Main/HomeViewModel.swift [hunk 3]
```diff
@@ -86,8 +81,54 @@ final class HomeViewModel: HomeViewModelLifeCycle, HomeViewInteraction, HomeView
...
+    func fetchUnfinishedFeeding() {
+        guard
+            let snapshot = model.fetchFeedingSnapshot(),
+            (Date.now - 3600) < snapshot.feedStartingDate
+        else {
+            return
+        }
+        Task {
+            do {
+                let feedingPoint = try await self.model.fetchFeedingPoint(snapshot.pointId)
```
Comment (diantyn) [inline animeal/src/Flows/Main/Modules/Home/Main/HomeViewModel.swift:106]: weak self, cause Task captures self with a strong ref

File: animeal/src/Flows/Main/Modules/Home/Main/HomeViewModel.swift [hunk 4]
```diff
@@ -86,8 +81,54 @@ final class HomeViewModel: HomeViewModelLifeCycle, HomeViewInteraction, HomeView
...
+                        isUnfinishedFeeding: true
+                    )
+                )
+            }
+        }
+    }
+
+    func startFeeding(feedingPointId id: String) {
+        Task {
+            do {
```
Comment (diantyn) [inline animeal/src/Flows/Main/Modules/Home/Main/HomeViewModel.swift:127]: weak self

File: animeal/src/Flows/Main/Modules/Home/Main/HomeViewModel.swift [hunk 5]
```diff
@@ -86,8 +81,54 @@ final class HomeViewModel: HomeViewModelLifeCycle, HomeViewInteraction, HomeView
             let action = model.fetchFeedingAction(request: .autoCancelFeeding)
             onFeedingActionHaveBeenPrepared?(feedingActionMapper.mapFeedingAction(action))
         case .confirmCancelFeeding:
-            load()
-            model.processCancelFeeding()
+            Task {
+                do {
+                    _ = try await model.processCancelFeeding()
```
Comment (diantyn) [inline animeal/src/Flows/Main/Modules/Home/Main/HomeViewModel.swift]: nit: we can mark model's method as discardableResult to get rid of using `_ = `

**Review by diantyn — COMMENTED** [review]
weak needs to be added

File: animeal/src/Flows/Main/Modules/Home/Main/HomeViewModel.swift (hunk 1 above)
Comment (igorTimofiejczyk) [reply animeal/src/Flows/Main/Modules/Home/Main/HomeViewModel.swift]: Done

File: animeal/src/Flows/Main/Modules/Home/Main/HomeViewModel.swift (hunk 3 above)
Comment (igorTimofiejczyk) [reply animeal/src/Flows/Main/Modules/Home/Main/HomeViewModel.swift:106]: Done

File: animeal/src/Flows/Main/Modules/Home/Main/HomeViewModel.swift (hunk 4 above)
Comment (igorTimofiejczyk) [reply animeal/src/Flows/Main/Modules/Home/Main/HomeViewModel.swift:127]: Done

File: animeal/src/Flows/Main/Modules/Home/Main/HomeViewModel.swift (hunk 5 above)
Comment (igorTimofiejczyk) [reply animeal/src/Flows/Main/Modules/Home/Main/HomeViewModel.swift]: Done

**Review by diantyn — APPROVED** [review]

**Pull request merged**